core/src/main/java/com/google/zxing/pdf417/PDF417ResultMetadata.java

```diff
@@ -42,14 +56,26 @@ public void setFileId(String fileId) {
     this.fileId = fileId;
   }
 
+  /**
+   * @deprecated use dedicated already parsed fields
+   * @return always null
+   */
+  @Deprecated
   public int[] getOptionalData() {
-    return optionalData;
+    return null;
   }
 
+  /**
+   * @deprecated no replacement
+   */
+  @Deprecated
```

Comment:
Is it not possible to continue to support this at all? just leave it also populated? I'd hate to introduce any breaking changes.

Reply:
We could just populate the array after the parsing. But there are also no tests for that functionality. I'll give it a try.

Reply:
Yeah I'd leave the functionality. Deprecating it is OK. This kind of has a replacement, right? the new fields you're adding have some of the same info?

Reply:
Exactly they now consume the array, the behavior now is not exactly the same as it parses the fields and allows for multiple ones to be set. For symbols that were parsed before the data found in the array should be the same.